synthetics/test_schema.go

```diff
@@ -16,17 +16,18 @@ func makeTestSchema(mode schemaMode) map[string]*schema.Schema {
 		},
 		"name": {
 			Type:     schema.TypeString,
-			Optional: optionalOnCreate(mode),
+			Required: requiredOnCreate(mode),
 			Computed: computedOnRead(mode),
 		},
 		"type": {
 			Type:     schema.TypeString,
-			Optional: optionalOnCreate(mode),
+			Optional: true,
 			Computed: computedOnRead(mode),
 		},
 		"status": {
 			Type:     schema.TypeString,
-			Optional: optionalOnCreate(mode),
+			Optional: optionalOnRead(mode),
```

Comment:
I guess clarity because it is truly optional only on read.

Reply:
User cannot pass `status` on read operation, when using Test data source. It is always computed on read. So setting optional in that case does nothing and is not needed. Deleting this line would maximize clarity, IMO.

Reply:
An attribute can be computed and optional, which means if you do not pass it to TF configuration it will be computed. So your comment applies to this example, but it doesn't apply to all usages of `optionalOnReadMode()`

Reply:
Here's explanation: https://github.com/hashicorp/terraform/issues/21278

Reply:
Discussion on Slack conclusion: optionalOnRead(mode) and all its usages are not needed and will be removed.

Reply:
Addressed: 59eb58fc73429bfe32c7956e590a621e62d04ddc